File: Spring/les/src/main/java/com/jga/les/model/Produto.java
package com.jga.les.model;

import java.util.List;

import org.hibernate.annotations.Cascade;
import org.hibernate.annotations.CascadeType;

import com.fasterxml.jackson.annotation.JsonBackReference;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Data
@Entity
public class Produto {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    private String nome;

    private String codigo;

    private boolean unitario;

    @NotNull
    private Double preco;

    @NotNull
    private Double custo;

    @OneToMany(mappedBy = "produto", orphanRemoval = true)
    @Cascade(CascadeType.ALL)
    @JsonBackReference
    private List<CompraProduto> compraProdutos;
}

File: Flutter/les/lib/model/produto/produto.dart

import 'package:les/model/entity.dart';

class Produto extends Entity{

  String codigo;
  bool unitario;
  String nome;
  double preco;
  double custo;

  Produto({
    super.id,
    required this.codigo,
    required this.unitario,
    required this.nome,
    required this.preco,
    required this.custo});

  @override
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'codigo': codigo,
      'unitario': unitario,
      'nome': nome,
      'preco': preco,
      'custo': custo
    };
  }

  factory Produto.fromJson(Map<String, dynamic> json) =>
    Produto(id: json['id'], codigo: json['codigo'], unitario: json['unitario'], nome: json['nome'], preco: json['preco'], custo: json['custo']);
}